### PR TITLE
Implement factory and config for sampling processor (head-based)

### LINKED ISSUE #145: Implement factory and config for "sampling" processor (head-based)
See config.go/factory.go in "addattributesprocessor" package as an example.

### PR DESCRIPTION
**Pull request by MohamedElqdusy**
Fix #145

### CODE REVIEW

**Comment by MohamedElqdusy** [comment]
@songy23 I thought the same as your suggestion but I decided to follow @pjanotti [comment](https://github.com/open-telemetry/opentelemetry-service/pull/152#discussion_r303719166)

File: processor/tracesamplerprocessor/factory.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+	"github.com/open-telemetry/opentelemetry-service/config/configerror"
+	"github.com/open-telemetry/opentelemetry-service/config/configmodels"
+	"github.com/open-telemetry/opentelemetry-service/consumer"
+	"github.com/open-telemetry/opentelemetry-service/processor"
+)
+
+const (
+	// The value of "type" trace-sample in configuration.
+	typeStr = "trace-sample"
```
[the remark this answers is not in the document]
Comment (pjanotti) [reply processor/tracesamplerprocessor/factory.go]: That said in light of the conversation above I think it makes more sense to rename the package... let's do in a separate PR so we can review the names of all processors and tags. I will open an issue for that.

[the remark this answers is not in the document]
Comment (pjanotti) [reply processor/tracesamplerprocessor/factory.go]: https://github.com/open-telemetry/opentelemetry-service/issues/187

File: processor/tracesamplerprocessor/factory.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	return typeStr
+}
+
+// CreateDefaultConfig creates the default configuration for processor.
+func (f *Factory) CreateDefaultConfig() configmodels.Processor {
+	return &TraceSamplerCfg{
+		ProcessorSettings: configmodels.ProcessorSettings{
+			TypeVal: typeStr,
+			NameVal: typeStr,
+		},
```
[the remark this answers is not in the document]
Comment (MohamedElqdusy) [reply processor/tracesamplerprocessor/factory.go]: I think HashSeed already has a default value 0 "initialized implicitly", as It's of type "uint32". the same implies for SamplingPercentage "float32", isn't it?

[the remark this answers is not in the document]
Comment (ccaraman) [reply processor/tracesamplerprocessor/factory.go]: I'm not sure. My golang struct initialization knowledge is shallow. I'll do a bit of digging and see what I can find.

[the remark this answers is not in the document]
Comment (ccaraman) [reply processor/tracesamplerprocessor/factory.go]: From what I've found, they will be set to 0. So we should be good to go. 
My reference: https://programming.guide/go/default-zero-value.html

[the remark this answers is not in the document]
Comment (MohamedElqdusy) [reply processor/tracesamplerprocessor/factory.go]: Maybe you would be interested in a formal [ref](https://golang.org/ref/spec#The_zero_value)

[the remark this answers is not in the document]
Comment (MohamedElqdusy) [reply processor/tracesamplerprocessor/factory.go]: Maybe you are more interested in a formal [ref](https://golang.org/ref/spec#The_zero_value)

[the remark this answers is not in the document]
Comment (ccaraman) [reply processor/tracesamplerprocessor/factory.go]: Thank you for the formal ref!

**Pull request merged**